editor/src/clj/editor/gui.clj

```diff
@@ -661,11 +746,190 @@
 
+;; SDK api
+(defmacro layout-property-getter [prop-sym]
+  {:pre [(symbol? prop-sym)]}
+  (let [prop-kw (keyword prop-sym)]
+    `(g/fnk [~'prop->value ~prop-sym]
+       (get ~'prop->value ~prop-kw ~prop-sym))))
+
+(defn layout-property-set-fn [_evaluation-context node-id _old-value _new-value]
+  (concat
+    (g/invalidate-output node-id :prop->value)
+    (g/invalidate-output node-id :layout->prop->value)))
+
+;; SDK api
+(defmacro layout-property-setter [prop-sym]
+  {:pre [(symbol? prop-sym)]}
+  layout-property-set-fn)
+
+(defn- layout-property-prop-def? [prop-def]
+  (let [value-dependencies (-> prop-def :value :dependencies)]
+    (and (= 2 (count value-dependencies))
+         (contains? value-dependencies :prop->value))))
+
+(defn- node-type-deref->stripped-prop-kws-raw [node-type-deref]
+  {:pre [(in/node-type-deref? node-type-deref)]}
+  (->> (:property node-type-deref)
+       (e/keep
+         (fn [[prop-kw prop-def]]
+           (when-not (layout-property-prop-def? prop-def)
+             prop-kw)))
+       (sort)
+       (vec)))
+
+(def ^:private node-type-deref->stripped-prop-kws (fn/memoize node-type-deref->stripped-prop-kws-raw))
+
+(defn- make-prop->value-for-default-layout [node]
+  (let [node-properties (gt/own-properties node)]
+    (if (coll/empty? node-properties)
+      {}
+      (let [node-type (g/node-type node)
+            stripped-prop-kws (node-type-deref->stripped-prop-kws @node-type)]
+        (persistent!
+          (reduce
+            dissoc!
+            (transient node-properties)
+            stripped-prop-kws))))))
+
+(defn- make-recursive-prop->value-for-default-layout [basis gui-node-id]
+  (let [[root-node-id & override-node-ids] (g/override-originals basis gui-node-id)
+        root-node (g/node-by-id basis root-node-id)
+
+        node-properties
+        (reduce (fn [node-properties override-node-id]
+                  (if-let [override-node (g/node-by-id basis override-node-id)]
+                    (reduce conj! node-properties (gt/overridden-properties override-node))
+                    node-properties))
+                (transient (or (gt/own-properties root-node) {}))
+                override-node-ids)]
+
+    (if (coll/empty? node-properties)
+      {}
+      (let [node-type (g/node-type root-node)
+            stripped-prop-kws (node-type-deref->stripped-prop-kws @node-type)]
+        (persistent!
+          (reduce dissoc! node-properties stripped-prop-kws))))))
+
+(defn- update-layout-property [evaluation-context node-id prop-kw update-fn & args]
+  (let [old-value (g/node-value node-id prop-kw evaluation-context)
+        new-value (apply update-fn old-value args)
+        current-layout (g/node-value node-id :current-layout evaluation-context)]
+    (if (str/blank? current-layout)
+      (g/set-property node-id prop-kw new-value)
+      (g/update-property
+        node-id :layout->prop->override
+        update current-layout
+        assoc prop-kw new-value))))
```

Comment:
This is the mechanism for updating the value of a layout property. We use this function in particular in our custom implementations of the transform manipulator `defmethods`. Moving a node in the scene view should update its pose in the `current-layout` only, after all.